### ISSUE TITLE
Bug Bash: 3.11.0

### ISSUE THREAD

**Issue by CsCherrYY** [badsyntax/vscode-gradle#1191]
cc: @Eskibear 
Test project: https://github.com/spring-projects/spring-petclinic

- [x] https://github.com/microsoft/vscode-gradle/pull/1174

steps to verify:

1. Open the spring-petclinic project
2. Open `Gradle projects` view, double click a specific task
3. Can see the task executing

- [x] https://github.com/microsoft/vscode-gradle/pull/1169

steps to verify:

1. Open the spring-petclinic project
2. change setting `java.jdt.ls.java.home` to a specific jdk path and keep `java.import.gradle.java.home` being unset
3. after reloading window, open `Gradle projects` view, check the output channel named `Gradle for Java`, can see a log with the following format:
> [info] Java Home: ${your configured jdk path in `java.jdt.ls.java.home`}
4. change setting `java.import.gradle.java.home` to a different jdk from the one in `java.jdt.ls.java.home`
5. Click refresh button in the `Gradle projects` view, check the output channel named `Gradle for Java`, can see a log with the following format:
> [info] Java Home: ${your configured jdk path in `java.import.gradle.java.home`}

- [x] https://github.com/microsoft/vscode-gradle/pull/1182

steps to verify:

1. Open the spring-petclinic project
2. In `Gradle` Side Bar view, can see there is no `Pinned Tasks` view
3. Right click the `test` task from `verification` task configuration, choose `Pin Task`
4. Can see a new tasks folder under the same project in the view, named `Pinned Tasks`
5. Can find the task just pinned in this folder
6. Right click the pinned task, can see the following buttons: 
  - Run Task (With Args) 
  - Debug Task (With Args)
  - Pin Task With Args
  - Unpin Task
  - Open build file

**Comment by Eskibear** [comment]
Verified. 

> Click refresh button in the Gradle projects view

Yes, only after I click refresh, the new jdk is shown in logs. But I do see some logs generated once I changed the settings. I believe you've already listened to the settings change. So why not make it automatic?

**Comment by CsCherrYY** [comment]
previously there was a bug about not showing view content, see https://github.com/microsoft/vscode-gradle/issues/903, the root cause is two refresh requests having race condition will cause empty result of `vscode.TreeView.getChildren()`. It's an upstream issue: https://github.com/microsoft/vscode/issues/129019

So in https://github.com/microsoft/vscode-gradle/pull/917 we just use a workaround to solve that issue, simplify remove the `refresh()` hook when the client connects to the server, that also results in the non-automatic refresh when reconnecting the server. Now I see the upstream issue has been resolved and I just tried and found it works. 

@Eskibear My suggestion here is that we can roll back the workaround in https://github.com/microsoft/vscode-gradle/pull/917 in `develop` branch and let it in the pre-release version to see if this error still exists or not. For the stable version, we can release it this time without the roll back and keep monitoring the pre-release version.
